exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillFilterAggregateTransposeRule.java

```diff
@@ -24,20 +24,29 @@
 import org.apache.calcite.rel.core.Filter;
 import org.apache.calcite.rel.core.RelFactories;
 import org.apache.calcite.rel.rules.FilterAggregateTransposeRule;
+import org.apache.calcite.tools.RelBuilderFactory;
 import org.apache.drill.exec.planner.DrillRelBuilder;
 
 public class DrillFilterAggregateTransposeRule extends FilterAggregateTransposeRule{
 
   // Since Calcite's default FilterAggregateTransposeRule would match Filter on top of Aggregate, it potentially will match Rels with mixed CONVENTION trait.
-  // Here override match method, such that the rule matchs with Rel in the same CONVENTION.
+  // Here override match method, such that the rule matches with Rel in the same CONVENTION.
 
   public static final FilterAggregateTransposeRule INSTANCE = new DrillFilterAggregateTransposeRule();
 
+  public static final FilterAggregateTransposeRule DRILL_LOGICAL_INSTANCE = new DrillFilterAggregateTransposeRule(
+      DrillRelBuilder.proto(DrillRelFactories.DRILL_LOGICAL_FILTER_FACTORY, DrillRelFactories.DRILL_LOGICAL_AGGREGATE_FACTORY),
+      Filter.class, Aggregate.class);
+
   private DrillFilterAggregateTransposeRule() {
     super(Filter.class, DrillRelBuilder.proto(Contexts.of(RelFactories.DEFAULT_FILTER_FACTORY)),
         Aggregate.class);
   }
 
+  private DrillFilterAggregateTransposeRule(RelBuilderFactory relBuilderFactory,
```

Comment:
Let's combine two constructors into one:
`private DrillFilterAggregateTransposeRule(RelBuilderFactory relBuilderFactory) {`
`    super(Filter.class, relBuilderFactory, Aggregate.class);`
`}`
In result you can use this constructor for `INSTANCE` and `DRILL_LOGICAL_INSTANCE`, but with different `RelBuilderFactory`